components/layout/Layout.js

```diff
@@ -1,40 +1,105 @@
 import { useRouter } from 'next/router'
-import DocsLayout from './DocsLayout'
+import DocsLayout, {buildNavTree} from './DocsLayout'
 import BlogLayout from './BlogLayout'
 import JobsLayout from './JobsLayout'
 import GenericPostLayout from './GenericPostLayout'
 import BlogContext from '../../context/BlogContext'
 import JobsContext from '../../context/JobsContext'
-import { getPostBySlug, getAllPosts } from '../../lib/api'
+import { getPostBySlug, getAllPosts, getDocBySlug } from '../../lib/api'
 
 export default function Layout({ children }) {
   const { pathname } = useRouter()
+  const posts = getAllPosts()
+  const allDocPosts = posts.filter(p => p.slug.startsWith('/docs/'))
+  const treePosts = buildNavTree(allDocPosts)
+  let structuredPosts = []
+  const convertDocPosts = (docObject) => {
```

Comment:
for future us 😄 , please add a comment what the purpose of this function is, why the conversion is needed and what is converted. I don't think you are converting anything, you are rather generating a flat list of all docs 🤔

Reply:
Sure :+1: